src/simulator.ts

```diff
@@ -15,6 +16,52 @@ declare global {
   }
 }
 
+function initServiceWorker() {
+  window.addEventListener("load", () => {
+    if ("serviceWorker" in navigator) {
+      navigator.serviceWorker.register("sw.js").then(
+        (registration) => {
+          console.log("Simulator service worker registration successful");
+          // Reload the page when a new service worker is installed.
+          registration.onupdatefound = function () {
+            const installingWorker = registration.installing;
+            if (installingWorker) {
+              installingWorker.onstatechange = function () {
+                if (
+                  installingWorker.state === "installed" &&
+                  navigator.serviceWorker.controller
+                ) {
+                  window.location.reload();
+                }
+              };
+            }
+          };
+        },
+        (error) => {
+          console.error(
+            `Simulator service worker registration failed: ${error}`
+          );
+        }
+      );
+    } else {
+      console.error("Service workers are not supported.");
+    }
+  });
+}
+
+if (flags.sw) {
+  initServiceWorker();
+} else {
+  navigator.serviceWorker.getRegistrations().then(registrations => {
```

Comment:
It seems odd to carefully check `if ("serviceWorker" in navigator)` above and then assume it here.

Reply:
You could just move the check to be around the if/else I guess.

Reply:
Done in [a1c8cff](https://github.com/microbit-foundation/micropython-microbit-v2-simulator/pull/115/commits/a1c8cff6d0c45f0ef0c750d4585fcd2432c7aac5).